lib/maze/reflective_servlet.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module Maze
+  # Receives HTTP requests and responds according to the parameters given, which are:
+  # - delay_ms - milliseconds to wait before responding
+  # - status - HTTP response code
+  # For GET requests these are expected to passed as GET parameters,
+  # for POST requests they are expected to be given as JSON fields.
+  class ReflectiveServlet < WEBrick::HTTPServlet::AbstractServlet
+
+    # Accepts a GET request to provide a reflective response to.
+    #
+    # @param request [HTTPRequest] The incoming GET request
+    # @param response [HTTPResponse] The response to return
+    def do_GET(request, response)
+      delay_ms = request.query['delay_ms']
+      status = request.query['status']
+      reflect response, delay_ms, status
+    end
+
+    # Accepts a POST request to provide a reflective response to.
+    #
+    # @param request [HTTPRequest] The incoming GET request
+    # @param response [HTTPResponse] The response to return
+    def do_POST(request, response)
+      body = JSON.parse(request.body)
```

Comment:
It may be worth having this check the content-type first, as other services may end up sending other mime-types (such as form-data) to test things like build/symbolication handling

Reply:
Do you mean so that it returns a reasonable response body and status code for unsupported content types, rather than a horrible 500?  Makes sense...

Reply:
Guard now added.